### PR TITLE
feat(EMS-1696-1697): Reactivate an account

### PR DESCRIPTION
**Pull request by ttbarnes**
This PR adds the ability for a user to reactivate their account by consuming a token/link sent via email.

## Changes

- Add E2E test coverage for:
  - Blocking an account, attempting to reactivate and visit the "verify email" page with an expired token.
  - Visiting the "verify email" page directly with an expired token.
  - Visiting the "verify email" page directly with no token.
  - Blocking an account and visiting the "verify email" with a valid token rediects to the "account is reactivated" page (happy path).
  - "Reactivated" page.
  - Signing into an account and getting to the dashboard with an account that has been reactivated after being blocked.
- Add `reactivationHash` to GQL queries made in E2E tests. This is used to consume a valid token in E2E tests.
- Create a new GQL resolver, `verifyAccountReactivationToken`. This resolver:
  - Checks if an account exists.
  - Checks if the reactivation token has expired.
  - Update the account so that `isBlocked` becomes false, `isVerified` becomes true and the reactivation hash and expiry are nullified.
  - Delete all authentication retries that belong to the account.
- Create a new UI GET routes, controllers and pages:
  - "Account suspended - verify email": consume a token from query params and calls the API. redirects appropriately.
  - "Account suspended - link expired": Simple page. Note - this currently only has a page title due to design/dev issues that need to be discussed/resolved
  - "Account suspended - reactivated": Simple page.
- Add the "account reactivated" route to a list of routes in the `applicationAccess` middleware.

## Other improvements

- Update API field ID constants.
- Improve documentation for `DATE_5_MINUTES_FROM_NOW` API constant.
- Fix some typos.
- Improve some logging.
- Bump cypress version.

### CODE REVIEW

File: src/ui/server/controllers/insurance/account/suspended/verify-email/index.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  try {
+    const { token } = req.query;
+
+    if (!token) {
+      return res.redirect(SUSPENDED_ROOT);
+    }
+
+    const response = await api.keystone.account.verifyAccountReactivationToken(token);
+
+    if (!response.success || response.expired) {
```
Comment (Zainzzkk) [inline src/ui/server/controllers/insurance/account/suspended/verify-email/index.ts:32]: is it worth conditional chaining these?

File: e2e-tests/cypress/e2e/journeys/insurance/account/suspended/link-expired/account-suspended-link-expired.spec.js [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+      const { data } = accountsResponse.body;
+
+      const [firstAccount] = data.accounts;
+      const account = firstAccount;
+
+      /**
+       * Update the account's reactivation expiry date via the API,
+       * so that we can mimic missing the verification period.
+       */
+      const now = new Date();
```
Comment (Zainzzkk) [inline e2e-tests/cypress/e2e/journeys/insurance/account/suspended/link-expired/account-suspended-link-expired.spec.js:55]: could we put 55 to 62 in a helper? easier to read and could be reused?

Comment (ttbarnes) [reply e2e-tests/cypress/e2e/journeys/insurance/account/suspended/link-expired/account-suspended-link-expired.spec.js:55]: Woud like to do after this PR please, there's some other instances of this and various other API functions that need to be moved into helpers

**Review by Zainzzkk — DISMISSED** [review]

File: src/ui/server/controllers/insurance/account/suspended/verify-email/index.ts (hunk 1 above)
Comment (ttbarnes) [reply src/ui/server/controllers/insurance/account/suspended/verify-email/index.ts:32]: It'd correctly fail or be undefined as it's only 1 level deep, we have this in other places as well

**Review by Zainzzkk — APPROVED** [review]

**Review by Zainzzkk — APPROVED** [review]

File: e2e-tests/cypress/e2e/journeys/insurance/application-submission/submit-an-application-has-not-traded-with-buyer-and-no-code-of-conduct.spec.js [hunk 3]
```diff
@@ -10,6 +10,19 @@ context('Insurance - submit an application without `have traded with buyer befor
   let url;
 
   before(() => {
+    cy.createAccount({});
+  //   cy.completeSignInAndSubmitAnApplication({
```
Comment (Zainzzkk) [inline e2e-tests/cypress/e2e/journeys/insurance/application-submission/submit-an-application-has-not-traded-with-buyer-and-no-code-of-conduct.spec.js:14]: can this be removed in next PR?

Comment (ttbarnes) [reply e2e-tests/cypress/e2e/journeys/insurance/application-submission/submit-an-application-has-not-traded-with-buyer-and-no-code-of-conduct.spec.js:14]: Good spot, yes

**Pull request merged**